jdaviz/configs/imviz/plugins/line_profile_xy/line_profile_xy.py

```diff
@@ -105,44 +104,24 @@ def vue_draw_plot(self, msg={}):
             self.reset_results()
             return
 
-        xy_limits = viewer._get_zoom_limits(data)
-        x_limits = xy_limits[:, 0]
-        y_limits = xy_limits[:, 1]
-        x_min = x_limits.min()
-        x_max = x_limits.max()
-        y_min = y_limits.min()
-        y_max = y_limits.max()
-
         comp = data.get_component(data.main_components[0])
         if comp.units:
             y_label = comp.units
         else:
             y_label = 'Value'
 
         self.plot_across_x.figure.title = f'X={x}'
-        line_x = self.plot_across_x.marks['line']
-        line_x.x, line_x.y = range(comp.data.shape[0]), comp.data[:, x]
-        line_x.scales['x'].min, line_x.scales['x'].max = y_min, y_max
+        self.plot_across_x._update_data('line', x=range(comp.data.shape[0]), y=comp.data[:, x],
+                                        reset_lims=True)
+        self.plot_across_x.update_style('line', line_visible=True,
+                                        markers_visible=False, color='gray')
         self.plot_across_x.figure.axes[1].label = y_label
 
-        y_min = max(int(y_min), 0)

```

Comment:
This block and similar block below that were deleted need to be restored but in your new way, so the Line Profile plots respect image zoom limits.